src/main/java/com/es/phoneshop/web/ProductDetailsPageServlet.java

```diff
@@ -0,0 +1,37 @@
+package com.es.phoneshop.web;
+
+import com.es.phoneshop.exception.ProductNotFoundException;
+import com.es.phoneshop.model.product.Product;
+import com.es.phoneshop.service.CustomProductService;
+import com.es.phoneshop.service.ProductService;
+import jakarta.servlet.ServletConfig;
+import jakarta.servlet.ServletException;
+import jakarta.servlet.http.HttpServlet;
+import jakarta.servlet.http.HttpServletRequest;
+import jakarta.servlet.http.HttpServletResponse;
+
+import java.io.IOException;
+
+public class ProductDetailsPageServlet extends HttpServlet {
+
+    private ProductService productService;
+
+    @Override
+    public void init(ServletConfig config) throws ServletException {
+        super.init(config);
+        productService = CustomProductService.getInstance();
+    }
+
+    @Override
+    protected void doGet(HttpServletRequest req, HttpServletResponse resp) throws ServletException, IOException {
+        String code = req.getPathInfo().substring(1);
+        try {
+            Product product = productService.getProductByCode(code);
+            req.setAttribute("product", product);
+            req.getRequestDispatcher("/WEB-INF/pages/productDetails.jsp").forward(req, resp);
+        } catch (ProductNotFoundException productNotFoundException) {
```

Comment:
such runtime-s are not required to be caught, you can define error-page which will be bind to this type of exception